src/univariate/affine.jl

```
@@ -0,0 +1,332 @@
+    end
+    return AffineDistribution(μ, Diagonal(Ones(μ)), ρ)
+end
+
+
+"""
+    *(σ, dist)
+
+Return a version of `dist` that has been scaled by `σ`, where `σ` can be any real number or 
+matrix.
+"""
+function Base.:*(σ::Real, ρ::UnivariateDistribution)
+    if iszero(σ)
+        throw(ArgumentError("scale must be non-zero"))
+    end
+    return AffineDistribution(zero(σ), σ, ρ)
+end
+
+function Base.:*(σ::Real, ρ::Distribution)
+    if iszero(σ)
+        throw(ArgumentError("scale must be non-zero"))
+    end
+    return AffineDistribution(Zeros{eltype(ρ)}(size(ρ)), σ, ρ)
+end
+
+function Base.:*(σ::AbstractMatrix{<:Real}, ρ::MultivariateDistribution)
+    if iszero(σ)
+        throw(ArgumentError("scale must be non-zero"))
+    elseif size(σ)[end] ≠ size(ρ)[1]
+        throw(DimensionMismatch(
+            "scaling matrix and distribution have incompatible dimensions" *
+            "$(size(σ)) and $(size(ρ)))"
+        ))
+    end
+    return AffineDistribution(Zeros{eltype(ρ)}(σ), σ, ρ)
+end
+Base.:*(σ::AbstractVector{<:Real}, ρ::MultivariateDistribution) = Diagonal(σ) * ρ
+
+
+# Composing affine distributions
+
+function Base.:+(μ::Real, d::UnivariateAffine)
+    return AffineDistribution(μ + d.μ, d.σ, d.ρ)
+end
+function Base.:+(μ::AbstractArray, d::MultivariateAffine)
+    return AffineDistribution(μ + d.μ, d.σ, d.ρ)
+end
+
+
+function Base.:*(σ::Real, d::UnivariateAffine)
+    return AffineDistribution(σ * d.μ, σ * d.σ, d.ρ)
+end
+
+function Base.:*(σ::Real, d::MultivariateAffine)
+    return AffineDistribution(σ * d.μ, σ * d.σ, d.ρ)
+end
+
+function Base.:*(σ::AbstractMatrix, d::MultivariateAffine)
+    return AffineDistribution(σ * d.μ, σ * d.σ, d.ρ)
+end
+
+
+Base.:+(d::NonMatrixDistribution, μ::ArrayLike) = μ + d
+Base.:-(d::NonMatrixDistribution) = -1 * d
+Base.:-(d::NonMatrixDistribution, μ::ArrayLike) = -μ + d
+Base.:-(μ::ArrayLike, d::NonMatrixDistribution) = μ + -d
```

This is too general.

---

What is a `NonMatrixDistribution`?

---

`NonMatrix` is defined as either Univariate or Multivariate.

The generality shouldn't make a difference because all of these methods just hand off to another method, so they should throw the same error message and make the same checks if the combination isn't appropriate.